source/material_model/interface.cc

```diff
@@ -393,6 +399,85 @@ namespace aspect
       reaction_terms(n_points, std::vector<double>(n_comp, numbers::signaling_nan<double>()))
     {}
 
+    namespace internal
+    {
+      template <int dim>
+      void check(const std::vector<double> &quantity,
+                 const char *name,
+                 const unsigned int n_points)
+      {
+        Assert(quantity.size() == n_points,
+               ExcMessage("Quantity '"+std::string(name)+"' has an incorrect size!"));
+        for (unsigned int i=0; i<n_points; ++i)
+          Assert(!std::isnan(quantity[i]),
+                 ExcMessage("NaN in '"+std::string(name)+"' discovered."));
+      }
+
+      template <int dim>
+      void check(const std::vector<SymmetricTensor<4,dim> > &quantity,
+                 const char *name,
+                 const unsigned int n_points)
+      {
+        Assert(quantity.size() == n_points,
+               ExcMessage("Quantity '"+std::string(name)+"' has an incorrect size!"));
+        for (unsigned int i=0; i<n_points; ++i)
+          Assert(!std::isnan(quantity[i].norm()),
+                 ExcMessage("NaN in '"+std::string(name)+"' discovered."));
+      }
+
+      template <int dim>
+      void check(const std::vector<std::vector<double> > &quantity,
+                 const char *name,
+                 const unsigned int n_points,
+                 const unsigned int n_components)
+      {
+        Assert(quantity.size() == n_points,
+               ExcMessage("Quantity '"+std::string(name)+"' has an incorrect size!"));
+        for (unsigned int i=0; i<n_points; ++i)
+          {
+            Assert(quantity[i].size()==n_components,
+                   ExcMessage("Quantity '"+std::string(name)+"' has an incorrect number of components!"));
+            for (unsigned int c=0; c<n_components; ++c)
+              Assert(!std::isnan(quantity[i][c]),
+                     ExcMessage("NaN in '"+std::string(name)+"' discovered."));
+          }
+      }
+    }
+
+    template <int dim>
+    void
+    MaterialModelOutputs<dim>::validate() const
+    {
+#ifdef DEBUG
+#ifdef ASPECT_USE_FP_EXCEPTIONS
+      // disable floating point exceptions so we can report nicer errors:
+      fedisableexcept(FE_DIVBYZERO|FE_INVALID);
```

Comment:
You need this because you want to deal with signaling NaNs, right? Maybe say so in the comment.